homeassistant/components/light/hyperion.py

```diff
@@ -90,34 +140,107 @@ def turn_on(self, **kwargs):
         """Turn the lights on."""
         if ATTR_RGB_COLOR in kwargs:
             self._rgb_color = kwargs[ATTR_RGB_COLOR]
-        else:
+            self._rgb_mem = kwargs[ATTR_RGB_COLOR]
+        elif self._rgb_mem == [0, 0, 0]:
             self._rgb_color = self._default_color
+            self._rgb_mem = self._default_color
+        else:
+            self._rgb_color = self._rgb_mem
+
+        if ATTR_BRIGHTNESS in kwargs:
+            self._brightness = kwargs[ATTR_BRIGHTNESS]
 
+        if ATTR_EFFECT in kwargs:
+            self._effect = kwargs[ATTR_EFFECT]
+            if self._effect == 'HDMI':
+                self.json_request({'command': 'clearall'})
+                self._icon = 'mdi:video-input-hdmi'
+                self._brightness = 255
+                self._rgb_color = [125, 125, 125]
+                self._skip_check = True
+            else:
+                self.json_request({
+                    'command': 'effect',
+                    'priority': self._priority,
+                    'effect': {'name': self._effect}
+                })
+                self._icon = 'mdi:lava-lamp'
+                self._rgb_color = [175, 0, 255]
+                self._skip_check = True
+            return
+
+        cal_color = [int(round(x*float(self._brightness)/255))
+                     for x in self._rgb_color]
         self.json_request({
             'command': 'color',
             'priority': self._priority,
-            'color': self._rgb_color
+            'color': cal_color
         })
 
     def turn_off(self, **kwargs):
         """Disconnect all remotes."""
         self.json_request({'command': 'clearall'})
+        self.json_request({
+            'command': 'color',
+            'priority': self._priority,
+            'color': [0, 0, 0]
+        })
         self._rgb_color = [0, 0, 0]
```

Comment:
Same as above.

Reply:
I agree, I will update in a commit in a second